### ISSUE TITLE
cabal build using snaplet-persistent failing.

### ISSUE THREAD

**Issue by nuncanada** [Soostone/snaplet-persistent#10]
Trying to build with snap-0.9 leads do cabal not being able to satisfy dependencies, it complains about lens.

With snap-1.0 from github repository i got further, but being a newbie in Haskell I am having problems solving this:

```
src/Site.hs:117:32:
    No instance for (MonadLogger (Initializer App PersistState))
      arising from a use of ‘initPersist’
    In the second argument of ‘($)’, namely
      ‘initPersist (runMigrationUnsafe migrateAuth)’
    In a stmt of a 'do' block:
      d <- nestSnaplet "db" db
           $ initPersist (runMigrationUnsafe migrateAuth)
    In the second argument of ‘($)’, namely
      ‘do { h <- nestSnaplet "" heist $ heistInit "templates";
            s <- nestSnaplet "sess" sess
                 $ initCookieSessionManager "site_key.txt" "sess" (Just 3600);
            d <- nestSnaplet "db" db
                 $ initPersist (runMigrationUnsafe migrateAuth);
            a <- nestSnaplet "auth" auth
                 $ initPersistAuthManager sess (persistPool $ view snapletValue d);
            .... }’



omni.cabal:
Name:                omni
Version:             0.1
Synopsis:            Project Synopsis Here
Description:         Project Description Here
License:             BSD3
Author:              Flavio Botelho
Maintainer:          fezsentido@gmail.com
Stability:           Experimental
Category:            Web
Build-type:          Simple
Cabal-version:       >=1.2

Flag development
  Description: Whether to build the server in development (interpreted) mode
  Default: False

Flag old-base
  default: False
  manual: False

Executable omni
  hs-source-dirs: src
  main-is: Main.hs

  Build-depends:
    bytestring                >= 0.9.1   && < 0.11,
    heist                     >= 1.0     && < 1.1,
    map-syntax                >= 0.2     && < 0.3,
    monad-control             >= 0.3     && < 0.4,
    monad-logger	            >= 0.3     && < 0.4,
    mtl                       >= 2       && < 3,
    snap                      >= 1.0     && < 1.1,
    snap-core                 >= 1.0     && < 1.1,
    snap-server               >= 1.0     && < 1.1,
    snap-loader-static        >= 1.0     && < 1.1,
    text                      >= 0.11    && < 1.3,
    time                      >= 1.1     && < 1.6,
    transformers              >= 0.4,
    xmlhtml                   >= 0.1,
    esqueleto                 >= 2.1,
    persistent                >= 2.1,
    persistent-postgresql     >= 2.1,
    snaplet-persistent        == 0.4.1.1

  if flag(old-base)
    build-depends:
      base                      >= 4        && < 4.4,
      lens                      >= 3.7.6    && < 3.8
  else
    build-depends:
      base                      >= 4.4      && < 5,
      lens                      >= 3.7.6    && < 4.7

  if flag(development)
    build-depends:
      snap-loader-dynamic >= 1.0 && < 1.1
    cpp-options: -DDEVELOPMENT
    -- In development mode, speed is already going to suffer, so skip
    -- the fancy optimization flags.  Additionally, disable all
    -- warnings.  The hint library doesn't give an option to execute
    -- compiled code when there were also warnings, so disabling
    -- warnings allows quicker workflow.
    ghc-options: -threaded -w
  else
    if impl(ghc >= 6.12.0)
      ghc-options: -threaded -Wall -fwarn-tabs -funbox-strict-fields -O2
                   -fno-warn-orphans -fno-warn-unused-do-bind
    else
      ghc-options: -threaded -Wall -fwarn-tabs -funbox-strict-fields -O2
                   -fno-warn-orphans

```

**Comment by mightybyte** [comment]
I just released snaplet-persistent-0.5 to hackage.  If that doesn't fix your problem, please reopen the ticket.

**Comment by nuncanada** [comment]
Thanks a lot for the work. It will work as I started using eryx's fork and
it is working.
